### PR TITLE
Support media deep links

### PR DESCRIPTION
**Pull request by zwarm**
This PR adds support for media deeplinks as a part of the Super fast media uploads
- `wordpress.com/media/{domain name}`
- `intent://media/#Intent;scheme=jetpack;package=com.jetpack.android`

## To Test:
**Preparing**
- Remove other versions of the app from your device (WP and JP)
- Download the links zip file, unzip and update the test-media-links.html file to include your domain name.
[test-media-links.html.zip](https://github.com/wordpress-mobile/WordPress-Android/files/13640040/test-media-links.html.zip)
- Save the update html file and download to the device you will be using to test this PR
- Download and install JP app from this PR
- Login

**Testing the deep links**
- Open the downloaded `test-media-links.html` in a browser on your device
- Tap on the `Intent Media` link 
- ✅ Verify the app is launched and the media view is visible for the selected site
- Tap on the `wordpress.com/media` link
- ✅ Verify the app is launched and the media view is visible for the selected site
- Tap on the `wordpress.com/media/nonsense` link
- ✅ Verify the app is launched and the media view is visible for the selected site
- Tap on the `wordpress.com/media/..` for your domain link
- ✅ Verify the app is launched and the media view is visible for your domain

-----

## Regression Notes

1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

-----

## PR Submission Checklist:
- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/deeplinks/handlers/MediaLinkHandler.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+class MediaLinkHandler
+@Inject constructor(private val deepLinkUriUtils: DeepLinkUriUtils) : DeepLinkHandler {
+    /**
+     * Returns true if the URI looks like `wordpress.com/media`
+     */
+    override fun shouldHandleUrl(uri: UriWrapper): Boolean {
+        return  (uri.host == HOST_WORDPRESS_COM &&
+                uri.pathSegments.firstOrNull() == MEDIA_PATH) || uri.host == MEDIA_PATH
+    }
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/deeplinks/handlers/MediaLinkHandler.kt:22]: There was a typo in the MEDIA_PATH variable, I fixed it with the commit b0b034cf64e4a4cbe3ec94d5b4768b5f29998e22

File: WordPress/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -477,6 +467,19 @@
                 android:host="wordpress.com"
                 android:pathPattern="/notifications/.*"
                 android:scheme="http" />
+
+                <data
+                    android:host="wordpress.com"
+                    android:pathPattern="/media/..*"
+                    android:scheme="http" />
+
```
Comment (AjeshRPai) [inline WordPress/src/main/AndroidManifest.xml:475]: Hey @zwarm 
When I was testing the changes, `wordpress.com/media` deeplink was not working for me. I added a path pattern of  `android:pathPattern="/media/.*"` to fix that. Commit 05189a62eeb62cfa8a634a77697060c342628330. Hope thats ok.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java [hunk 3]
```diff
@@ -926,6 +930,12 @@ private void handleOpenPageIntent(@NonNull Intent intent) {
                 case ARG_BLOGGING_PROMPTS_ONBOARDING:
                     showBloggingPromptsOnboarding();
                     break;
+                case ARG_MEDIA:
+                    if (!mSelectedSiteRepository.hasSelectedSite()) {
+                        initSelectedSite();
+                    }
+                    mActivityNavigator.viewCurrentBlogMedia(this, getSelectedSite());
+                    break;
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:938]: Hey @zwarm 

I updated logic to use `ActivityNavigator` here. I think we should gradually move all the Activity Navigation logic to `ActivityNavigator`. Commit a0d27d189db58f21f4aa5be3896247917f87ce37

File: WordPress/src/main/java/org/wordpress/android/ui/deeplinks/DeepLinkNavigator.kt [hunk 4]
```diff
@@ -114,5 +117,7 @@ class DeepLinkNavigator
         object OpenLoginPrologue : NavigateAction()
         data class OpenJetpackForDeepLink(val action: String?, val uri: UriWrapper) : NavigateAction()
         object OpenJetpackStaticPosterView : NavigateAction()
+        data class OpenMediaForSite(val site: SiteModel) : NavigateAction()
+        object OpenMedia : NavigateAction()
```
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/deeplinks/DeepLinkNavigator.kt:121]: Updated the logic to use `ActivityNavigator` here. Commit -  61fc7694f3413e5eb92524256c5b726c07698794

**Review by AjeshRPai — APPROVED** [review]
Hey @zwarm 

I have tested and is working as expected. I made some minor changes. I am approving but not merging so that you can take a look at the changes I have made and revert if necessary.

File: WordPress/src/main/AndroidManifest.xml (hunk 2 above)
Comment (zwarm) [reply WordPress/src/main/AndroidManifest.xml:475]: That's weird ... we don't support .* for pages or stats or others.

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java (hunk 3 above)
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java:938]: Great idea - I forgot `ActivityNavigator` existed.

**Pull request merged**